### DISCUSSION TITLE
Custom field type object 2way visibility

### DISCUSSION THREAD

**Discussion by kirillmuravyev** [netbox-community/netbox#12714] [Ideas]
Hi All.
I was playing with custom fields a little bit and noticed interesting thing that might be useful to implement.
I have creates a custom field with type "object" in "dcim -> device", object type is "virtual machine".
So I want to link some devices to virtual machines.

BTW, there is no way to model virtual network devices deployed on cloud VMs, when you have a virtual machine in cloud and network device software deployed on it. And we treat those things differently, vurtual machine itself is supported by one team, but virtual network device installed on that machine is supported by another team.
That is why I want to link device and virtual machine together.

So, when I did the above relation, I can open device and link it to virtual machine (and I see that relationship), but from the virtual machine perspective I do not see that relation:
**Custom field in DEvice view (I see the link to virtual machine):**
![image](https://github.com/netbox-community/netbox/assets/40544508/2da3261f-797e-44e2-a94b-11d1313e4cc2)

But when you open virtual machine you do not see that object relates to this one

I think would be good to see that relation automatically if I reference object in custom field. Otherwise you have to create 2 custom fields and maintain them in order to track it from both sides

**Comment by mtinberg** [comment]
Having a Script which runs periodically and updates a reverse relationship is a way to solve this using the tools built into the existing system, but the way I've solved this (from before object refs were a thing) is to make a Custom URL on the object where you want the reverse relation, and encode the search for the primary relation, eg. on VM have a URL which templates

/dcim/device/?cf_runs_on=$my_vm_id_here

then you have one click to see what relation is.  It's not as neat as having it on the main object page itself, but it requires even less to maintain.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Kirill Muravyev ***@***.***>
Sent: Thursday, May 25, 2023 7:28 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Custom field type object 2way visibility (Discussion #12714)


Hi All.
I was playing with custom fields a little bit and noticed interesting thing that might be useful to implement.
I have creates a custom field with type "object" in "dcim -> device", object type is "virtual machine".
So I want to link some devices to virtual machines.

BTW, there is no way to model virtual network devices deployed on cloud VMs, when you have a virtual machine in cloud and network device software deployed on it. And we treat those things differently, vurtual machine itself is supported by one team, but virtual network device installed on that machine is supported by another team.
That is why I want to link device and virtual machine together.

So, when I did the above relation, I can open device and link it to virtual machine (and I see that relationship), but from the virtual machine perspective I do not see that relation:
Custom field in DEvice view (I see the link to virtual machine):
[image]<https://user-images.githubusercontent.com/40544508/240911413-2da3261f-797e-44e2-a94b-11d1313e4cc2.png>

But when you open virtual machine you do not see that object relates to this one

I think would be good to see that relation automatically if I reference object in custom field. Otherwise you have to create 2 custom fields and maintain them in order to track it from both sides

**Comment by kirillmuravyev** [comment]
Thanks @mtinberg  for the idea!
Trying to do the Custom Link within virtual machine tab, but failed to make it work properly ) trying further

**Comment by kirillmuravyev** [comment]
Found the way with custome URL:

`dcim/devices/?cf_runson={{ object.id }}`

My custom field name is "runson".
You can check the url using the filter in "devices", it uses id instead of name
